Core/Settings/SpeedLimiter.cs

```diff
@@ -0,0 +1,49 @@
+﻿using Godot;
+using RoverControlApp.Core.JSONConverters;
+using System;
+using System.Text.Json.Serialization;
+
+namespace RoverControlApp.Core.Settings;
+
+[JsonConverter(typeof(SpeedLimiterConverter))]
+public partial class SpeedLimiter : SettingBase, ICloneable
+{
+
+	public SpeedLimiter()
+	{
+		_enabled = false;
+		_maxSpeed = 0.5f;
+	}
+
+	public SpeedLimiter(bool enabled, float maxSpeed)
+	{
+		_enabled = enabled;
+		_maxSpeed = maxSpeed;
+	}
+
+	public object Clone()
+	{
+		return new SpeedLimiter()
+		{
+			Enabled = _enabled,
+			MaxSpeed = _maxSpeed
+		};
+	}
+
+	[SettingsManagerVisible(cellMode: TreeItem.TreeCellMode.Check)]
+	public bool Enabled
+	{
+		get => _enabled;
+		set => EmitSignal_SettingChanged(ref _enabled, value);
+	}
+
+	[SettingsManagerVisible(cellMode: TreeItem.TreeCellMode.Range,  formatData: "0.2;1;0.05;f;f", customName: "MaxSpeed (multiplier)")]
```

Comment:
While I can see your point with the maximum value of the multipier, I think that we should see how the rover actually drives at up to 5% vs 20% throttle. If it cannot move with a minimum multiplier applied why should we implement it all (on the other hand it would be very safe indeed xd).

Reply:
yup tru